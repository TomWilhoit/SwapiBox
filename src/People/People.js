import React, { Component } from "react";
import "../People/People.scss";
import { fetchData } from "../utils/API";

class People extends Component {
  constructor() {
    super();
  }

  componentDidMount() {
    const url = "https://swapi.co/api/people";
    fetchData(url)
      .then(peopleData =>
        this.setState({
          people: peopleData.results
        })
      )
      .catch(error => error.message);
  }

  returnPeople = () => {
    let personItem = this.props.peopleArray.map((person, index) => {
      return (
        <div key={index} className="card">
          <p>Name: {person.personName}</p>
          <p>Homeworld: {person.planetName}</p>
          <p>Population: {person.population}</p>
        </div>
      );
    });
    return personItem;
  };

  render() {
    if (this.props.peopleArray.length === 0) {
      return <div className="loading">Loading</div>;
    } else {
      return (
        <div className="People">
          <h1 className="title">People</h1>
          <div className="button-container" buttons>
            <button className="btn" onClick={this.props.goToMain}>
              MAIN
            </button>
            <button className="btn" onClick={this.props.goToPlanets}>
              PLANETS
            </button>
            <button className="btn" onClick={this.props.goToVehicles}>
              VEHICLES
            </button>
          </div>
          <div className="card-container">{this.returnPeople()}</div>
        </div>
      );
    }
  }
}

export default People;
