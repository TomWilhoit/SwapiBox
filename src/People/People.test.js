import React from "react";
import People from "../People/People";
import { shallow } from "enzyme";

describe("People", () => {
  let wrapper;
  let peopleArray = [1, 2, 3]
  

  beforeEach(() => {
    wrapper = shallow(<People peopleArray={peopleArray} />)

  });

  it("should match snapshot when all data is passed correctly", () => {
    
    expect(wrapper).toMatchSnapshot();
  });
});