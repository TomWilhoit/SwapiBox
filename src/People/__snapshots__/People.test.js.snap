// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match snapshot when all data is passed correctly 1`] = `
<div
  className="People"
>
  <h1
    className="title"
  >
    People
  </h1>
  <div
    className="button-container"
  >
    <button
      className="btn"
    >
      MAIN
    </button>
    <button
      className="btn"
    >
      PLANETS
    </button>
    <button
      className="btn"
    >
      VEHICLES
    </button>
  </div>
  <div
    className="card-container"
  >
    <div
      className="card"
      key="0"
    >
      <p>
        Name: 
        Jim
      </p>
      <p>
        Homeworld: 
      </p>
      <p>
        Population: 
      </p>
    </div>
    <div
      className="card"
      key="1"
    >
      <p>
        Name: 
        Frank
      </p>
      <p>
        Homeworld: 
      </p>
      <p>
        Population: 
      </p>
    </div>
    <div
      className="card"
      key="2"
    >
      <p>
        Name: 
        Harry
      </p>
      <p>
        Homeworld: 
      </p>
      <p>
        Population: 
      </p>
    </div>
  </div>
</div>
`;
